### DISCUSSION TITLE
Dynamic Badge showing 'Invalid'

### DISCUSSION THREAD

**Discussion by treeben77** [badges/shields#9407] [Q&A] [answered]
I'm trying to make a badge that shows the number of likes on a discourse post, but most of the time it says 'Invalid'. The strange thing is, sometimes it works. This is the badge im trying to make:

![DevForum Post](https://img.shields.io/badge/dynamic/json?url=https%3A%2F%2Fdevforum.roblox.com%2Ft%2Frblx-open-cloud-roblox-open-cloud-wrapper-for-python%2F1991959%2F1.json&query=%24.like_count&suffix=%20likes&style=for-the-badge&logo=robloxstudio&logoColor=white&label=DevForum%20Post&labelColor=%23009fff&color=%23353535&cacheSeconds=86400)

Badge URL: https://img.shields.io/badge/dynamic/json?url=https%3A%2F%2Fdevforum.roblox.com%2Ft%2Frblx-open-cloud-roblox-open-cloud-wrapper-for-python%2F1991959%2F1.json&query=%24.like_count&suffix=%20likes&style=for-the-badge&logo=robloxstudio&logoColor=white&label=DevForum%20Post&labelColor=%23009fff&color=%23353535&cacheSeconds=86400

JSON Source: https://devforum.roblox.com/t/1991959.json

This is what it looks like when it does load properly:
![devforum post badge](https://github.com/badges/shields/assets/77905642/292af857-da32-413a-ad38-b2c0a2b6763c)

I found another post that said they might have blocked the user agent, but when I tested `Shields.io/2003a` it seemed to work just fine. I'm suspecting the problem could be because the forum can be slow to load and maybe there's a load timeout.

Any help is appreciated!

**Comment by calebcartwright** [comment]
The badge currently appears fine ATM so hard to take a closer look, but if the message simply says "Invalid" it means that Shields is getting an unexpected/unsupported http response from the json endpoint you have specified in the `?url=...` query parameter.

If I had to guess I'd say that there's some intermittent status of the file as it's being updated which results in some jsonpath processing error with the query you've provided, as at least at a cursory glance of the code on our side, that seems like the most likely path that would produce an "invalid" error without any additional context.

**Comment by treeben77** [comment, marked as answer]
Nevermind turns out how I tested the user agent was overridden with the service I was using, and seemed to be going through. I'm really sorry for wasting your time @calebcartwright.
